include/cpr/session.h

```diff
@@ -230,8 +231,10 @@ class Session : public std::enable_shared_from_this<Session> {
     friend MultiPerform;
 
 
-    bool hasBodyOrPayload_{false};
     bool chunkedTransferEncoding_{false};
+    std::optional<cpr::Payload> payload_;
+    std::optional<cpr::Body> body_;
+    std::optional<cpr::Multipart> multipart_;
```

Comment:
Jup, it's an ABI breaking change and no I do not care since we never state that increments in the patch version would provide a stable ABI.

But thanks for noticing!